File: src/main/java/com/oracle/ocmjea/gof/behavioral/visitor/ComputerPartDisplayVisitor.java
package com.oracle.ocmjea.gof.behavioral.visitor;

public class ComputerPartDisplayVisitor implements ComputerPartVisitor {

	@Override
	public void visit(Computer computer) {
		System.out.println("Mostrando Computer.");
	}

	@Override
	public void visit(Mouse mouse) {
		System.out.println("Mostrando Mouse.");
	}

	@Override
	public void visit(Teclado teclado) {
		System.out.println("Mostrando Keyboard.");
	}

	@Override
	public void visit(Monitor monitor) {
		System.out.println("Mostrando Monitor.");
	}
}

File: design-patterns/src/main/kotlin/designpatterns/behavioral/visitor/ComputerPartDisplayVisitor.kt
package designpatterns.behavioral.visitor

class ComputerPartDisplayVisitor : ComputerPartVisitor {
    override fun visit(computer: Computer) = println("Displaying Computer.")
    override fun visit(mouse: Mouse) = println("Displaying Mouse.")
    override fun visit(keyboard: Keyboard) = println("Displaying Keyboard.")
    override fun visit(monitor: Monitor) = println("Displaying Monitor.")
}
